sympy/calculus/accumulationbounds.py

```diff
@@ -523,6 +524,8 @@ def __pow__(self, other):
                             return AccumBounds(self.max**other, oo)
                         if self.max.is_zero:
                             return AccumBounds(self.min**other, oo)
+                        if self.has(Symbol):
+                            return AccumBounds(self.max**other, self.min**other)
```

Comment:
Are the arguments correctly ordered this way?

Reply:
Certainly... , we're just trying to replicate the general case i.e.
```
>>> y = Symbol('y', positive=True)
>>> expr = y + AccumBounds(-1, 1)
>>> lst = [2, 3]
>>> for i in lst:
...     term = expr.subs(y, i)
...     print(term**-1)
...
AccumBounds(1/3, 1)
AccumBounds(1/4, 1/2)
```
I was also thinking about something which could work better than what we have currently . The `__rtruediv__` method and `__pow__` for positive powers are quite consistent with the results they return and I feel we could let these methods handle the general cases. I would expect this diff to maybe return more consistent results.
```
--- a/sympy/calculus/accumulationbounds.py
+++ b/sympy/calculus/accumulationbounds.py
@@ -524,9 +524,7 @@ def __pow__(self, other):
                             return AccumBounds(self.max**other, oo)
                         if self.max.is_zero:
                             return AccumBounds(self.min**other, oo)
-                        if self.has(Symbol):
-                            return AccumBounds(self.max**other, self.min**other)
-                        return AccumBounds(0, oo)
+                        return (1/self)**(-other)
                     return AccumBounds(
                         S.Zero, Max(self.min**other, self.max**other))
                 elif other % 2 == 1:
@@ -535,9 +533,7 @@ def __pow__(self, other):
                             return AccumBounds(self.max**other, oo)
                         if self.max.is_zero:
                             return AccumBounds(-oo, self.min**other)
-                        if self.has(Symbol):
-                            return AccumBounds(self.max**other, self.min**other)
-                        return AccumBounds(-oo, oo)
+                        return (1/self)**(-other)
                     return AccumBounds(self.min**other, self.max**other)
```
This also ensures consistent outputs /errors being returned overall
```
>>> x = Symbol('x')
>>> y = Symbol('y', positive=True)
>>> expr1 = y + AccumBounds(-1, 1)
>>> expr2 = y + AccumBounds(2, 4)
>>> expr3 = y + AccumBounds(-4, -2)
>>> expr1
AccumBounds(y - 1, y + 1)
>>> expr2
AccumBounds(y + 2, y + 4)
>>> expr3
AccumBounds(y - 4, y - 2)
>>> 1/expr1
Traceback (most recent call last):
....
    raise ValueError("Only real AccumulationBounds are supported")
ValueError: Only real AccumulationBounds are supported
>>> expr1**-1
Traceback (most recent call last):
....
    raise ValueError("Only real AccumulationBounds are supported")
ValueError: Only real AccumulationBounds are supported
>>> 1/expr2
AccumBounds(Min(1/(y + 2), 1/(y + 4)), Max(1/(y + 2), 1/(y + 4)))
>>> expr2**-1
AccumBounds(Min(1/(y + 2), 1/(y + 4)), Max(1/(y + 2), 1/(y + 4)))
>>> 1/expr3
Traceback (most recent call last):
....
    raise ValueError("Only real AccumulationBounds are supported")
ValueError: Only real AccumulationBounds are supported
>>> expr3**-1
Traceback (most recent call last):
....
    raise ValueError("Only real AccumulationBounds are supported")
ValueError: Only real AccumulationBounds are supported
```

Reply:
This change is being made with respect to this `limit(log(x)/(log(x) + sin(x)), x, oo)` where gruntz should fail and we get the result from simple heuristics . Implementing the diff above would give the exact traceback as we have on master 
```
>>> gruntz(log(x)/(log(x) + sin(x)), x, oo)
Traceback (most recent call last):
.....................
  File "C:\Users\91989\sympy\sympy\sympy\calculus\accumulationbounds.py", line 194, in __new__
    raise ValueError("Only real AccumulationBounds are supported")
ValueError: Only real AccumulationBounds are supported

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\91989\sympy\sympy\sympy\core\mul.py", line 1907, in _eval_nseries
    raise ValueError
ValueError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
.........................
  File "C:\Users\91989\sympy\sympy\sympy\calculus\accumulationbounds.py", line 194, in __new__
    raise ValueError("Only real AccumulationBounds are supported")
ValueError: Only real AccumulationBounds are supported
```
 Hence I see this would maintain consistency with whatever we have on master rather than introducing anything new. I'll commit this and you could let me know if you think otherwise .